docs/kubernetes/crds/agentendpoint.mdx

```diff
@@ -0,0 +1,389 @@
+Unlike `AgentEndpoint` resources, the `CloudEndpoint` resource gives you the ability to configure whether pooling should be allowed or not.
+
+**Type:** `string`
+
+**Required:** yes
+
+**Default:** none
+
+The following formats are accepted:
+
+- **Domain**
+  - When using the domain format you are only defining the domain. The scheme and port will be inferred.
+  - example: `example.org`
+- **Origin**
+  - When using the origin format you are defining the protocol, domain and port. HTTP endpoints accept ports 80 or 443 with respective protocol.
+  - examples:
+    - `https://example.ngrok.app`
+    - `https://example.ngrok.app:443`
+    - `tcp://1.tcp.ngrok.io:12345`
+    - `tls://example.ngrok.app`
+- **Scheme (shorthand)**
+  - When using scheme you are defining the protocol and will receive back a randomly assigned ngrok address.
+  - examples:
+    - `https://`
+    - `tcp://`
+    - `tls://`
+    - `http://`
+- **Empty**
+  - When empty your endpoint will default to be https and receive back a randomly assigned ngrok address.
+- **Internal**
+  - When ending your url with .internal, an internal endpoint will be created. Internal Endpoints cannot be accessed directly, but rather can only be accessed using the forward-internal traffic policy action.
+  - example: `some.domain.internal`
+
+### `spec.upstream`
+
+Configures the upstream for the `AgentEndpoint` that should reveive traffic sent to the `AgentEndpoint`
+
+**Type:** `Object`
+
+**Required:** yes
+
+**Default:** none
+
+**Fields:**
+
+| Field Name                                        | Type            | Required | Default | Description                                                              |
+| ------------------------------------------------- | --------------- | -------- | ------- | ------------------------------------------------------------------------ |
+| [`spec.upstream.url`](#specupstreamurl)           | `string`        | yes      | none    | The local or remote address that incoming traffic should be forwarded to |
+| [`spec.upstream.protocol`](#specupstreamprotocol) | `string` (enum) | no       | `http1` | The protocol to use when connecting to the upstream                      |
+
+### `spec.upstream.url`
+
+The local or remote address you would like to incoming traffic to be forwarded to.
+
+**Type:** `string`
+
+**Required:** yes
+
+**Default:** none
+
+Accepted formats are:
+
+- **Origin**
+  - When using the origin format you are defining the protocol, domain and port.
+    When no port is present and scheme is https or http the port will be inferred.
+    - For https port will be `443`
+    - For http port will be `80`
+  - examples:
+    - `https://example.org`
+    - `http://example.org:80`
+    - `tcp://127.0.0.1:80`
+- **Domain**
+  - This is only allowed for https and http endpoints.
+    For tcp and tls endpoints host and port is required.
```

Comment:
```suggestion
    For TCP and TLS endpoints host and port is required.
```

From what I have seen around our docs, protocols seem to be capitalized. I'll just drop 1 of these suggestions so as not to pollute your PR with a bunch of noise